moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -0,0 +1,737 @@
+    ASSERT_TRUE(getParam("num_ik_tests", num_ik_tests_));
+    ASSERT_TRUE(getParam("num_ik_multiple_tests", num_ik_multiple_tests_));
+    ASSERT_TRUE(getParam("num_nearest_ik_tests", num_nearest_ik_tests_));
+
+    ASSERT_TRUE(robot_model_->hasJointModelGroup(group_name_));
+    ASSERT_TRUE(robot_model_->hasLinkModel(root_link_));
+    ASSERT_TRUE(robot_model_->hasLinkModel(tip_link_));
+  }
+
+public:
+  auto createUniqueInstance(const std::string& name) const
+  {
+    return kinematics_loader_->createUniqueInstance(name);
+  }
+
+  static const SharedData& instance()
+  {
+    static SharedData instance_;
+    return instance_;
+  }
+  static void release()
+  {
+    SharedData& shared = const_cast<SharedData&>(instance());
+    shared.kinematics_loader_.reset();
+  }
+};
+
+class KinematicsTest : public ::testing::Test
+{
+protected:
+  void operator=(const SharedData& data)
+  {
+    robot_model_ = data.robot_model_;
+    jmg_ = robot_model_->getJointModelGroup(data.group_name_);
+    root_link_ = data.root_link_;
+    tip_link_ = data.tip_link_;
+    group_name_ = data.group_name_;
+    joints_ = data.joints_;
+    seed_ = data.seed_;
+    consistency_limits_ = data.consistency_limits_;
+    timeout_ = data.timeout_;
+    num_fk_tests_ = data.num_fk_tests_;
+    num_ik_cb_tests_ = data.num_ik_cb_tests_;
+    num_ik_tests_ = data.num_ik_tests_;
+    num_ik_multiple_tests_ = data.num_ik_multiple_tests_;
+    num_nearest_ik_tests_ = data.num_nearest_ik_tests_;
+  }
+
+  void SetUp() override
+  {
+    *this = SharedData::instance();
+
+    std::string plugin_name;
+    ASSERT_TRUE(getParam("ik_plugin_name", plugin_name));
+    ROS_INFO_STREAM("Loading " << plugin_name);
+    kinematics_solver_ = SharedData::instance().createUniqueInstance(plugin_name);
+    ASSERT_TRUE(bool(kinematics_solver_)) << "Failed to load plugin: " << plugin_name;
+
+    // initializing plugin
+    ASSERT_TRUE(kinematics_solver_->initialize(*robot_model_, group_name_, root_link_, { tip_link_ },
+                                               DEFAULT_SEARCH_DISCRETIZATION) ||
+                kinematics_solver_->initialize(ROBOT_DESCRIPTION_PARAM, group_name_, root_link_, { tip_link_ },
+                                               DEFAULT_SEARCH_DISCRETIZATION))
+        << "Solver failed to initialize";
+
+    jmg_ = robot_model_->getJointModelGroup(kinematics_solver_->getGroupName());
+    ASSERT_TRUE(jmg_);
+
+    // Validate chain information
+    ASSERT_EQ(root_link_, kinematics_solver_->getBaseFrame());
+    ASSERT_FALSE(kinematics_solver_->getTipFrames().empty());
+    ASSERT_EQ(tip_link_, kinematics_solver_->getTipFrame());
+    ASSERT_EQ(joints_, kinematics_solver_->getJointNames());
+  }
+
+public:
+  testing::AssertionResult isNear(const char* expr1, const char* expr2, const char* abs_error_expr,
+                                  const geometry_msgs::Point& val1, const geometry_msgs::Point& val2, double abs_error)
```

Comment:
These seems pretty generic, could we move it to [robot_model_test_utils.h](https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_core/utils/include/moveit/utils/robot_model_test_utils.h) ?

Reply:
Done.

Reply:
Thanks for trying this. 

Why can't we install gtest? Testing is important to software and shouldn't be held back due to the desire to reduce dependencies

Reply:
Moved convo to issue: https://github.com/ros-planning/moveit/issues/1323